projects/kit/components/input-phone/input-phone.component.ts

```diff
@@ -195,6 +195,7 @@ export class TuiInputPhoneComponent
 
         this.updateSearch(parsed);
         this.value = parsed === this.countryCode || isText(parsed) ? '' : parsed;
+        this.value = this.value === '' && !this.allowText ? this.countryCode : this.value;
```

Comment:
Is it ok that we trigger value setter twice? This would probably update control twice, can we do it in one go?

Reply:
fixed